libs/components/indicators/src/lib/modules/illustration/illustration.component.html

```diff
@@ -1,10 +1,12 @@
-<img
-  alt=""
-  loading="lazy"
+<div
   [attr.data-sky-illustration-name]="name()"
-  [class]="'sky-illustration-img sky-illustration-img-' + size()"
-  [ngClass]="{
-    'sky-illustration-img-loaded': !!url()
-  }"
-  [src]="url()"
-/>
+  [class]="'sky-illustration-wrapper sky-illustration-' + size()"
+>
+  @if (svgHref()) {
+    <svg aria-hidden="true" class="sky-illustration-svg">
+      <use [attr.xlink:href]="svgHref()" />
+    </svg>
+  } @else {
+    <img alt="" loading="lazy" class="sky-illustration-img" [src]="url()" />
+  }
```

Comment:
You can use a variable to store the result of the signal to avoid calling it twice.

```suggestion
  @if (svgHref(); as svgHrefValue) {
    <svg aria-hidden="true" class="sky-illustration-svg">
      <use [attr.xlink:href]="svgHrefValue" />
    </svg>
  } @else {
    <img alt="" loading="lazy" class="sky-illustration-img" [src]="url()" />
  }
```